custom/icds/repeaters/generators/phi.py

```diff
@@ -0,0 +1,70 @@
+import json
+
+from django.core.serializers.json import DjangoJSONEncoder
+
+from corehq import toggles
+from corehq.apps.hqcase.utils import update_case
+from corehq.motech.repeaters.repeater_generators import (
+    CaseRepeaterJsonPayloadGenerator,
+)
+
+
+class BasePayloadGenerator(CaseRepeaterJsonPayloadGenerator):
+    @staticmethod
+    def enabled_for_domain(domain):
+        return toggles.PHI_CAS_INTEGRATION.enabled(domain)
+
+
+class SearchByParamsPayloadGenerator(BasePayloadGenerator):
+    @staticmethod
+    def _gender(gender):
+        if gender:
+            if gender == 'male':
+                return 'M'
+            elif gender == 'female':
+                return 'F'
+        return ""
+
+    def get_payload(self, repeat_record, case):
+        data = {
+            "beneficaryname": case.name,
+            "fathername": case.get_case_property('fathers_name') or "",
+            "husbandname": case.get_case_property('husbands_name') or "",
+            "mothername": case.get_case_property('mothers_name') or "",
+            "gender": self._gender(case.get_case_property('gender')),
+            "villagecode": 442639,
+            "subdistrictcode": 3318,
+            "districtcode": 378,
+            "statecode": 22,
```

Comment:
keeping these values hard coded for now till we know how to fetch them.
I can't validate as of now if i can leave them as blank because the PHI Api isnt working, so keeping these values from the example in the document for now.

Reply:
Using a `ValueSource` subclass to get these values would give you the option of using location metadata for these codes, e.g. 
```javascript
    "statecode": CaseOwnerAncestorLocationField(
        location_field="phi_statecode"
    ).get_value(case_trigger_info),
```
where `case_trigger_info` is a [`CaseTriggerInfo`](https://github.com/dimagi/commcare-hq/blob/7194221f443f6b45d0600b31cb99b08fb1a90ec2/corehq/motech/value_source.py#L24) instance, and includes case data and other useful stuff.

Reply:
Thanks @kaapstorm .. these values are hard coded for now but will/should eventually come from case/location. Going to keep this note for later when these would actually get fetched